src/Interfaces/AbstractPolyhedron_functions.jl

```diff
@@ -476,39 +655,42 @@ function _linear_map_vrep(M::AbstractMatrix{N}, P::AbstractPolyhedron{N}) where
     return _linear_map_vrep(M, P)
 end
 
-function _linear_map_hrep(M::AbstractMatrix{N}, P::AbstractPolyhedron{N},
-                          use_inv::Bool;
-                          inverse::Union{Nothing, AbstractMatrix{N}}=nothing
-                         ) where {N<:Real}
-    constraints = _linear_map_hrep_helper(M, P, use_inv, inverse=inverse)
+# generic function for the AbstractPolyhedron interface => returns an HPolyhedron
+function _linear_map_hrep_helper(M::AbstractMatrix{N}, P::AbstractPolyhedron{N},
+                                 algo::AbstractLinearMapAlgorithm) where {N<:Real}
+    constraints = _linear_map_hrep(M, P, algo)
     return HPolyhedron(constraints)
 end
 
-function _linear_map_hrep_helper(M::AbstractMatrix{N}, P::AbstractPolyhedron{N},
-                                 use_inv::Bool;
-                                 inverse::Union{Nothing, AbstractMatrix{N}}=nothing
-                                ) where {N<:Real}
+# preconditions should have been checked in the callinear_map_hrep(M, P, algo)ler function
+function _linear_map_hrep(M::AbstractMatrix{N}, P::AbstractPolyhedron{N},
+                          algo::LinearMapInverse) where {N}
+    inverse = isempty(algo.inverse) ? inv(M) : algo.inverse
```

Comment:
What does `isempty(::Matrix)` mean? I think you want to remove the type constraint that `algo.inverse` has to be a matrix and instead assign the default value `nothing`.
Maybe even define a function `hasinverse(algo::LinearMapInverse)` for clarity.

Reply:
The reason was that the empty matrix is used as a check that the inverse has not been assigned yet; this is needed [here](https://github.com/JuliaReach/LazySets.jl/pull/1950/files#diff-5e796138f8958d1d7c93fd38b67eb60aR599, because the matrix is computed in the `_linear_map_hrep_helper`.

ou're right that it is a bit odd to use the struct that way. After a second thought, here is my new proposal:

- Observe that the `_check_algorithm_applies` functions do not actually need an instances of the algorithm, so we can dispatch on its type instead.
- Remove the line `inverse = isempty(algo.inverse) ? inv(M) : algo.inverse`, and compute it before hand.

I think that the logic is better in this way :+1:

Reply:
>     * Observe that the `_check_algorithm_applies` functions do not actually need an instance of the algorithm, so we can dispatch on its type instead.

I think that is what I meant: allow the type `Nothing` for `algo.inverse` and then you can dispatch on that.